.kokoro/docker/cloudbuild.yaml

```diff
@@ -0,0 +1,63 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+timeout: 7200s # 2 hours
+substitutions:
+  _JAVA_SHARED_CONFIG_VERSION: '1.5.7'
+  _GRAALVM_A: 'graalvm22_3_jdk11'
+  _GRAALVM_B: 'graalvm22_3_jdk17'
+steps:
+  # GraalVM A build
+  - name: gcr.io/cloud-builders/docker
+    args: ["build", "-t", "gcr.io/$PROJECT_ID/graalvm_a:${_JAVA_SHARED_CONFIG_VERSION}", "--file", "${_GRAALVM_A}.Dockerfile", "."]
+    env:
+    dir: .kokoro/docker
+    id: graalvm-a-build
+    waitFor: ["-"]
+  - name: gcr.io/gcp-runtimes/structure_test
+    args:
+      ["-i", "gcr.io/$PROJECT_ID/graalvm_a:${_JAVA_SHARED_CONFIG_VERSION}", "--config", ".kokoro/docker/${_GRAALVM_A}.yaml", "-v"]
+    waitFor: ["graalvm-a-build"]
+  - name: gcr.io/cloud-builders/docker
+    args:
+      [
+          "tag",
+          "gcr.io/$PROJECT_ID/graalvm_a:${_JAVA_SHARED_CONFIG_VERSION}",
+          "gcr.io/cloud-devrel-public-resources/graalvm_a:${_JAVA_SHARED_CONFIG_VERSION}",
+      ]
+    waitFor: ["graalvm-a-build"]
+
+  # GraalVM B build
+  - name: gcr.io/cloud-builders/docker
+    args: [ "build", "-t", "gcr.io/$PROJECT_ID/graalvm_b:${_JAVA_SHARED_CONFIG_VERSION}", "--file", "${_GRAALVM_B}.Dockerfile", "." ]
+    dir: .kokoro/docker
+    id: graalvm-b-build
+    waitFor: [ "-" ]
+  - name: gcr.io/gcp-runtimes/structure_test
+    args:
+      [ "-i", "gcr.io/$PROJECT_ID/graalvm_b:${_JAVA_SHARED_CONFIG_VERSION}", "--config", ".kokoro/docker/${_GRAALVM_B}.yaml", "-v" ]
+    waitFor: [ "graalvm-b-build" ]
+  - name: gcr.io/cloud-builders/docker
+    args:
+      [
+          "tag",
+          "gcr.io/$PROJECT_ID/graalvm_b:${_JAVA_SHARED_CONFIG_VERSION}",
+          "gcr.io/cloud-devrel-public-resources/graalvm_b:${_JAVA_SHARED_CONFIG_VERSION}",
+      ]
+    waitFor: [ "graalvm-b-build" ]
+images:
+  - gcr.io/$PROJECT_ID/graalvm_a:${_JAVA_SHARED_CONFIG_VERSION}
+  - gcr.io/cloud-devrel-public-resources/graalvm_a:${_JAVA_SHARED_CONFIG_VERSION}
+  - gcr.io/$PROJECT_ID/graalvm_b:${_JAVA_SHARED_CONFIG_VERSION}
+  - gcr.io/cloud-devrel-public-resources/graalvm_b:${_JAVA_SHARED_CONFIG_VERSION}
```

Comment:
@suztomo Wanted to call out these publishing destinations.  We're publishing to `gcr.io/cloud-devrel-kokoro-resources/` which will be used by our kokoro jobs downstream and `gcr.io/cloud-devrel-public-resources` which is a public repository of images for client library testing.

Reply:
Do you need both?

Reply:
`gcr.io/cloud-devrel-kokoro-resources` is used  by Kokoro jobs whereas `gcr.io/cloud-devrel-public-resources/` is used for making our testing images public to allow non-Googlers to run tests against the same image as us.  I've added this in here following convention in https://github.com/googleapis/testing-infra-docker. We seem to publish images to both locations for several other languages as well (nodejs, python).